### PR TITLE
Excalidraw update

### PR DESCRIPTION
**Pull request by hero101**
## Requires env variables for the collaboration service connection
- VITE_APP_COLLAB_URL
- VITE_APP_COLLAB_PATH
---
- requires the CSS to be imported independently

## Summary by CodeRabbit

- **New Features**
	- Updated dependency to the latest version of `@alkemio/excalidraw`, potentially introducing new features and improvements.
	- Added new environment variables for collaboration service URLs in the configuration.
	- Introduced a new optional `disabled` property for the `WhiteboardDialogTemplatesLibrary`, enhancing interactivity control.
	- Added a section in the README to clarify the usage of environment variables.

- **Bug Fixes**
	- Improved handling of incoming broadcast data and scene updates in collaboration, enhancing overall stability.
	- Enhanced user experience with clearer translations for whiteboard actions and confirmations.

- **Refactor**
	- Streamlined import paths for types from `@alkemio/excalidraw`, ensuring consistency and clarity in code organization.
	- Updated the `.gitignore` to exclude sensitive environment files.

- **Chores**
	- Removed unused environment variables and files, indicating a shift in the application’s architecture.
	- Added comments in the `.env` file for clarity on usage.

### CODE REVIEW

File: src/domain/collaboration/whiteboard/WhiteboardDialog/WhiteboardDialog.tsx [hunk 1]
```diff
@@ -30,19 +30,46 @@ import useLoadingState from '../../../shared/utils/useLoadingState';
...
+interface WhiteboardDialogProps {
   entities: {
-    whiteboard?: Whiteboard;
+    whiteboard?: WhiteboardDetails;
   };
   actions: {
     onCancel: () => void;
     onUpdate: (
-      whiteboard: Whiteboard,
+      whiteboard: WhiteboardDetails,
```
[the remark this answers is not in the document]
Comment (ccanos) [reply src/domain/collaboration/whiteboard/WhiteboardDialog/WhiteboardDialog.tsx:68]: for deleting we only need the id

File: src/domain/collaboration/whiteboard/WhiteboardDialog/WhiteboardDialog.tsx [hunk 2]
```diff
@@ -80,12 +106,7 @@ const useStyles = makeStyles(theme => ({
 
 type RelevantExcalidrawState = Pick<ExportedDataState, 'appState' | 'elements' | 'files'>;
 
-const WhiteboardDialog = <Whiteboard extends WhiteboardWithContent>({
-  entities,
-  actions,
-  options,
-  state,
-}: WhiteboardDialogProps<Whiteboard>) => {
+const WhiteboardDialog = ({ entities, actions, options, state }: WhiteboardDialogProps) => {
```
[the remark this answers is not in the document]
Comment (ccanos) [reply src/domain/collaboration/whiteboard/WhiteboardDialog/WhiteboardDialog.tsx:109]: we want to simplify the type for readability

File: src/domain/collaboration/whiteboard/WhiteboardDialog/WhiteboardDialog.tsx [hunk 3]
```diff
@@ -108,6 +129,7 @@ const WhiteboardDialog = <Whiteboard extends WhiteboardWithContent>({
   const columns = useGlobalGridColumns();
 
   const [lastSavedDate, setLastSavedDate] = useState<Date | undefined>(undefined);
+  const [isSceneInitialized, setSceneInitialized] = useState(false);
```
[the remark this answers is not in the document]
Comment (ccanos) [reply src/domain/collaboration/whiteboard/WhiteboardDialog/WhiteboardDialog.tsx:132]: setSceneInitialized is passed to `CollaborativeExcalidrawWrapper` and is called from inside there

File: src/domain/collaboration/whiteboard/WhiteboardDialog/WhiteboardDialog.tsx [hunk 4]
```diff
@@ -126,11 +148,11 @@ const WhiteboardDialog = <Whiteboard extends WhiteboardWithContent>({
   });
 
   const prepareWhiteboardForUpdate = async (
-    whiteboard: WhiteboardWithContent,
+    whiteboard: WhiteboardDetails,
     state: RelevantExcalidrawState | undefined,
     shouldUploadPreviewImages = true
   ): Promise<{
-    whiteboard: Whiteboard;
+    whiteboard: WhiteboardDetails;
```
[the remark this answers is not in the document]
Comment (ccanos) [reply src/domain/collaboration/whiteboard/WhiteboardDialog/WhiteboardDialog.tsx:155]: we'll do this in another moment

File: src/domain/collaboration/whiteboard/WhiteboardDialog/WhiteboardDialog.tsx [hunk 5]
```diff
@@ -30,19 +30,46 @@ import useLoadingState from '../../../shared/utils/useLoadingState';
...
+      url: string;
+      avatar?: { id: string; uri: string };
+    };
+  };
+}
+
+interface WhiteboardDialogProps {
   entities: {
-    whiteboard?: Whiteboard;
+    whiteboard?: WhiteboardDetails;
```
[the remark this answers is not in the document]
Comment (ccanos) [reply src/domain/collaboration/whiteboard/WhiteboardDialog/WhiteboardDialog.tsx]: addressed

**Review by ccanos — APPROVED** [review]

File: src/domain/collaboration/whiteboard/WhiteboardDialog/WhiteboardDialog.tsx [hunk 6]
```diff
@@ -30,19 +30,46 @@ import useLoadingState from '../../../shared/utils/useLoadingState';
...
+      url: string;
+      avatar?: { id: string; uri: string };
+    };
+  };
+}
+
+interface WhiteboardDialogProps {
   entities: {
-    whiteboard?: Whiteboard;
+    whiteboard: WhiteboardDetails | undefined;
```
[the remark this answers is not in the document]
Comment (ccanos) [reply src/domain/collaboration/whiteboard/WhiteboardDialog/WhiteboardDialog.tsx:63]: whiteboard is obtained from a query. making it mandatory complicates things in other components, it's better to handle undefined here

**Review by ccanos — APPROVED** [review]

**Review by valentinyanakiev — APPROVED** [review]

**Pull request merged**